rmf_api_msgs/samples/task_state/multi_dropoff_delivery.json

```diff
@@ -0,0 +1,214 @@
+          "location": "room_521",
+          "items": [
+            {
+              "type": "water",
+              "quantity": 1
+            }
+          ]
+        }
+      }
+    ],
+    "estimate_millis": 2000000
+  },
+  "completed": [
+    {
+      "phase": {
+        "id": 1,
+        "header": {
+          "category": "Pick Up",
+          "detail": {
+            "location": "Kitchen",
+            "items": [
+              {
+                "type": "soda",
+                "quantity": 1
+              },
+              {
+                "type": "water",
+                "quantity": 1
+              }
+            ]
+          },
+          "estimate_millis": 600000
+        },
+        "final_event_id": 0,
+        "events": [
+          {
+            "id": 0,
+            "status": "completed",
+            "name": "Sequence",
+            "detail": "",
+            "log": [],
+            "deps": [1, 2]
+          },
+          {
+            "id": 1,
+            "status": "completed",
+            "name": "Go to [place:kitchen]",
+            "detail": "",
+            "log": [
+              {
+                "tier": "info",
+                "unix_millis_time": 1636388415000,
+                "text": "Waiting for [door:kitchen_door] to open"
+              },
+              {
+                "tier": "info",
+                "unix_millis_time": 1636388422000,
+                "text": "Passed through [door:kitchen_door]"
+              },
+              {
+                "tier": "info",
+                "unix_millis_time": 1636388426000,
+                "text": "Arrived at [place:kitchen]"
+              }
+            ],
+            "deps": []
+          },
+          {
+            "id": 2,
+            "status": "completed",
+            "name": "Receive items",
+            "detail": [
+              {
+                "type": "soda",
+                "quantity": 1
+              },
+              {
+                "type": "water",
+                "quantity": 1
+              }
+            ],
+            "log": [
+              {
+                "tier": "info",
+                "unix_millis_time": 1636388430000,
+                "text": "Received [item:soda]"
+              },
+              {
+                "tier": "info",
+                "unix_millis_time": 1636388440000,
+                "text": "Received [item:water]"
+              }
+            ],
+            "deps": []
+          }
+        ]
+      }
+    }
+  ],
+  "active": {
+    "id": 2,
+    "header": {
+      "category": "Drop Off",
+      "detail": {
+        "location": "room_203",
+        "items": [
+          {
+            "type": "soda",
+            "quantity": 1
+          }
+        ]
+      },
+      "estimate_millis": 720000
+    },
+    "final_event_id": 0,
+    "events": [
+      {
+        "id": 0,
+        "status": "underway",
+        "name": "Sequence",
+        "detail": "",
+        "log": [],
+        "deps": [1, 2]
```

Comment:
It seems that the meaning of `deps` is dependent on `name`. As I understand it, a `Sequence`'s deps means that the events have to happen in sequence, according to the order its defined in `deps`, but this may be different for other type of events.

Or if `Sequence` is just a name, and that `deps` always means the events have to happen in order, then could we just remove this and have event `2` depend on event `1`?

Reason is because such an event is ambiguous to render in ui, we don't know if this event refers to some work or goal being done or is just a meta event reference other events. Even if we handle all known meta events, third party devs can add their own meta events which the ui will not be able to render correctly.

Reply:
You are correct that the exact meaning and behavior of the "dependencies" is dependent on the parent.

My intention is for the dependency hierarchy to be displayed as an expandable treeview. Each event in the sequence would be displayed as equals in the treeview hierarchy, and each of those events may have its own dependencies, which would be displayed as their children in the treeview. You can expand each item in the treeview to expand it and see its children to get more details about the event.

I have a bunch of ideas related to how to render this, but it will be difficult to explain in text. Maybe I'll jump into one of the web team meetings and we can workshop it.